scripts/waitForSync.py

```diff
@@ -0,0 +1,63 @@
+import os
+import subprocess
+import sys
+
+network = os.getenv("NETWORK")
+
+bad_logs = {"Exception": 1, "Missing node found!": 1}
+good_logs = {"Processed": 0, "Stats after finishing state": 0}
+required_count = {"Processed": 20, "Stats after finishing state": 1}
+
+if network not in {"joc-mainnet", "joc-testnet", "linea-mainnet", "linea-sepolia"}:
+    good_logs["Synced Chain Head"] = 0
+    required_count["Synced Chain Head"] = 20
+
+container_mapping = {
+    "base-": "sedge-execution-op-l2-client",
+    "op-": "sedge-execution-op-l2-client",
+}
+default_container_name = "sedge-execution-client"
+
+container_name = next(
+    (name for prefix, name in container_mapping.items() if network.startswith(prefix)),
+    default_container_name
+)
+
+process = subprocess.Popen(["docker", "logs", "-f", container_name], stdout=subprocess.PIPE, text=True)
+
+found_bad_log = False
+counter = 0
+
+try:
+    for line in process.stdout:
+        print(line.strip())
+
+        if found_bad_log:
+            counter += 1
+            if counter >= 100:
+                print("Exiting after capturing extra logs due to error.")
+                sys.exit(1)
+            continue
+
+        if any(bad_log in line for bad_log in bad_logs):
+            print(f"Error: Found bad log in line: {line.strip()}")
+            found_bad_log = True
+            continue
```

Comment:
Not sure I understand why it's implemented this way. From my understanding:

When a bad log is found, we'll set the flag to true;
After that, we'll run the for loop 100 more times, and then exit.

Is it used to capture extra logs?

Reply:
Yes exactly for this purpose